core/lib/vm/src/tracers/traits.rs

```diff
@@ -88,10 +81,22 @@ pub enum TracerExecutionStatus {
     Stop(TracerExecutionStopReason),
 }
 
-/// Stop the vm execution if the tracer conditions are met
-pub trait ExecutionEndTracer<H: HistoryMode> {
-    // Returns whether the vm execution should stop.
-    fn should_stop_execution(&self) -> TracerExecutionStatus {
-        TracerExecutionStatus::Continue
+impl TracerExecutionStatus {
+    /// Chose the stricter ExecutionStatus
+    /// If both statuses are Continue, then the result is Continue
+    /// If one of the statuses is Abort, then the result is Abort
+    /// If one of the statuses is Finish, then the result is Finish
```

Comment:
Nit: This part of the comment looks outdated.

Reply:
No..
We try to find both `Continue`, then If one of the statues abort than abort, then if one of them finish and neiter of them are abort, then it's finish..

Reply:
Ah, sorry; I've misread the comment. To be really nitpicky, `Abort` and `Finish` are stop *reasons* (judging by the type name), not *statuses*.